sql/src/main/java/io/crate/operation/reference/sys/node/NodeMemoryExpression.java

```diff
@@ -37,39 +37,51 @@
     public static final String USED_PERCENT = "used_percent";
     public static final String PROBE_TIMESTAMP = "probe_timestamp";
 
-    public NodeMemoryExpression(OsStats stats) {
-        addChildImplementations(stats);
+    public NodeMemoryExpression(final OsStats stats) {
+        addChildImplementations(stats.mem());
+        childImplementations.put(PROBE_TIMESTAMP, new SysNodeExpression<Long>() {
+            @Override
+            public Long value() {
+                return stats.timestamp();
+            }
+        });
     }
 
-    private void addChildImplementations(final OsStats stats) {
+    private void addChildImplementations(final OsStats.Mem mem) {
         childImplementations.put(FREE, new MemoryExpression() {
             @Override
             public Long value() {
-                return stats.mem().actualFree().bytes();
+                if (mem != null) {
```

Comment:
I wonder if it wouldn't be simpler to do the mem check once outside and in case it is null create one SysNodeExpression that returns -1  that is used for all columns